# Analyse AirBnB Data from Seattle - Udacity Data Sience Nanodegree

**Goal:**
Analyse data according the CRISP-DM (Cross Industry Standard Process for Data Mining) method.

**Data source:**

Seattle AirBnb Data is available in kaggle: https://www.kaggle.com/airbnb/seattle
- calendar.csv
- listings.csv
- reviews.csv

# 1. Business understanding

Interesting Questions on dataset:
 
a) What neighbourhoods are most booked in Seattle

b) What kind of seasonal effects exist on pricing? Correlation to occupancy ration in the whole city?

c) What is the impact of higher reviews on the number of bookings? What kind of impact factor is more important, price or reviews?

d)  What are the most occuring amenities? What effect do they have on number of bookings?

# 2. Data Understanding


In [98]:
# import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score, mean_squared_error


In [99]:
# read calendar.csv
df_cal = pd.read_csv('./data/calendar.csv')
print('Rows: ', df_cal.shape[0], 'Columns: ', df_cal.shape[1])
df_cal.head()

Rows:  1393570 Columns:  4


listing_id        date available   price
0      241032  2016-01-04         t  $85.00
1      241032  2016-01-05         t  $85.00
2      241032  2016-01-06         f     NaN
3      241032  2016-01-07         f     NaN
4      241032  2016-01-08         f     NaN

In [100]:
df_cal.dtypes

listing_id     int64
date          object
available     object
price         object
dtype: object

In [101]:
# read listings.csv
df_list = pd.read_csv('./data/listings.csv')
print('Rows: ', df_list.shape[0], 'Columns: ', df_list.shape[1])
df_list.head()


Rows:  3818 Columns:  92


id                           listing_url       scrape_id last_scraped  \
0   241032   https://www.airbnb.com/rooms/241032  20160104002432   2016-01-04   
1   953595   https://www.airbnb.com/rooms/953595  20160104002432   2016-01-04   
2  3308979  https://www.airbnb.com/rooms/3308979  20160104002432   2016-01-04   
3  7421966  https://www.airbnb.com/rooms/7421966  20160104002432   2016-01-04   
4   278830   https://www.airbnb.com/rooms/278830  20160104002432   2016-01-04   

                                  name  \
0         Stylish Queen Anne Apartment   
1   Bright & Airy Queen Anne Apartment   
2  New Modern House-Amazing water view   
3                   Queen Anne Chateau   
4       Charming craftsman 3 bdm house   

                                             summary  \
0                                                NaN   
1  Chemically sensitive? We've removed the irrita...   
2  New modern house built in 2013.  Spectacular s...   
3  A charming apartment that sits atop Queen Anne...   
4  Cozy family craftman house in beautiful neighb...   

                                               space  \
0  Make your self at home in this charming one-be...   
1  Beautiful, hypoallergenic apartment in an extr...   
2  Our house is modern, light and fresh with a wa...   
3                                                NaN   
4  Cozy family craftman house in beautiful neighb...   

                                         description experiences_offered  \
0  Make your self at home in this charming one-be...                none   
1  Chemically sensitive? We've removed the irrita...                none   
2  New modern house built in 2013.  Spectacular s...                none   
3  A charming apartment that sits atop Queen Anne...                none   
4  Cozy family craftman house in beautiful neighb...                none   

                               neighborhood_overview  ... review_scores_value  \
0                                                NaN  ...                10.0   
1  Queen Anne is a wonderful, truly functional vi...  ...                10.0   
2  Upper Queen Anne is a charming neighborhood fu...  ...                10.0   
3                                                NaN  ...                 NaN   
4  We are in the beautiful neighborhood of Queen ...  ...                 9.0   

  requires_license license jurisdiction_names instant_bookable  \
0                f     NaN         WASHINGTON                f   
1                f     NaN         WASHINGTON                f   
2                f     NaN         WASHINGTON                f   
3                f     NaN         WASHINGTON                f   
4                f     NaN         WASHINGTON                f   

  cancellation_policy  require_guest_profile_picture  \
0            moderate                              f   
1              strict                              t   
2              strict                              f   
3            flexible                              f   
4              strict                              f   

  require_guest_phone_verification calculated_host_listings_count  \
0                                f                              2   
1                                t                              6   
2                                f                              2   
3                                f                              1   
4                                f                              1   

  reviews_per_month  
0              4.07  
1              1.48  
2              1.15  
3               NaN  
4              0.89  

[5 rows x 92 columns]

In [102]:
# column names + types
list_type = df_list.dtypes
pd.set_option('display.max_rows', 100)
print(list_type)


id                                    int64
listing_url                          object
scrape_id                             int64
last_scraped                         object
name                                 object
summary                              object
space                                object
description                          object
experiences_offered                  object
neighborhood_overview                object
notes                                object
transit                              object
thumbnail_url                        object
medium_url                           object
picture_url                          object
xl_picture_url                       object
host_id                               int64
host_url                             object
host_name                            object
host_since                           object
host_location                        object
host_about                           object
host_response_time              

In [103]:
# read listings.csv
df_rev = pd.read_csv('./data/reviews.csv')
print('Rows: ', df_rev.shape[0], 'Columns: ', df_rev.shape[1])
df_rev.head()

Rows:  84849 Columns:  6


listing_id        id        date  reviewer_id reviewer_name  \
0     7202016  38917982  2015-07-19     28943674        Bianca   
1     7202016  39087409  2015-07-20     32440555         Frank   
2     7202016  39820030  2015-07-26     37722850           Ian   
3     7202016  40813543  2015-08-02     33671805        George   
4     7202016  41986501  2015-08-10     34959538          Ming   

                                            comments  
0  Cute and cozy place. Perfect location to every...  
1  Kelly has a great room in a very central locat...  
2  Very spacious apartment, and in a great neighb...  
3  Close to Seattle Center and all it has to offe...  
4  Kelly was a great host and very accommodating ...

In [104]:
df_rev.dtypes

listing_id        int64
id                int64
date             object
reviewer_id       int64
reviewer_name    object
comments         object
dtype: object

**Result of data understanding:**



| Table | No of rows | No of columns | Content|
| :-: | :-:| :-: | :-: |
| *calendar* | 1393570 | 4 | availability and price for each object depending on date |
| *listing* | 3818 | 92 |  information for each object (host, neighborhood, amenities, type) |
| *reviews* | 84849 | 9 | comments of reviewers for each object |

## 3. Data Preparation

**dataframe calendar:**
- convert datatypes
- remove NaN in column price
- extract year, month day 

In [105]:
# clean data in dataframe calendar
pd.set_option('display.max_row', 1000)
df_cal
#df_cal.dtypes

listing_id        date available   price
0            241032  2016-01-04         t  $85.00
1            241032  2016-01-05         t  $85.00
2            241032  2016-01-06         f     NaN
3            241032  2016-01-07         f     NaN
4            241032  2016-01-08         f     NaN
...             ...         ...       ...     ...
1393565    10208623  2016-12-29         f     NaN
1393566    10208623  2016-12-30         f     NaN
1393567    10208623  2016-12-31         f     NaN
1393568    10208623  2017-01-01         f     NaN
1393569    10208623  2017-01-02         f     NaN

[1393570 rows x 4 columns]

In [106]:
# convert data type of column available to boolean
df_cal['available'] = df_cal['available'] == 't'

# convert data type of column price to float
df_cal['price'] = df_cal['price'].str.replace('$', '')


df_cal['price'] = df_cal['price'].str.replace(',', '')
df_cal['price'] = df_cal['price'].astype(float)

# convert date to datatype datatime & extract year, day, month
df_cal['year'] = pd.DatetimeIndex(df_cal['date']).year
df_cal['day'] = pd.DatetimeIndex(df_cal['date']).day
df_cal['month'] = pd.DatetimeIndex(df_cal['date']).month
df_cal['date'] = pd.to_datetime(df_cal['date'])


In [107]:
# check if price is only NaN in case available is false
print(df_cal['available'].value_counts())
print(np.sum(df_cal.price.isna()))

# replace all na with 0
df_cal['price'].fillna(0, inplace=True)

True     934542
False    459028
Name: available, dtype: int64
459028


In [109]:
# final types
df_cal.dtypes

listing_id             int64
date          datetime64[ns]
available               bool
price                float64
year                   int64
day                    int64
month                  int64
dtype: object

**dataframe listing:**

- tbd
- tbd

## 4. Data Modeling
not necessary to answer business questions

## 5. Data Evaluation and Visualization